app/scripts/services/bindingModalUtils.js

```diff
@@ -0,0 +1,24 @@
+'use strict';
+
+angular.module('openshiftConsole')
+  .factory('BindingModalUtils', [
+    function() {
+      return {
+        sortApplications: function(deploymentConfigs,
+                                    deployments,
+                                    replicationControllers,
+                                    replicaSets,
+                                    statefulSets) {
+          // Don't waste time sorting on each data load, just sort when we have them all
+          if (deploymentConfigs && deployments && replicationControllers && replicaSets && statefulSets) {
+            var apiObjects = deploymentConfigs.concat(deployments)
+                                              .concat(replicationControllers)
+                                              .concat(replicaSets)
+                                              .concat(statefulSets);
```

Comment:
Probably want to check if these are arrays before calling `concat`. Or at least make a comment that the inputs must be arrays.